Investigate timeseries assertion error

19 of the 4159 cells potentially to be released in June fail feature extraction with the error below. Need to understand why we are getting this error, and if there is a way to generate the output nwb file with extracted features, and maybe a tag indicating sweeps that fall outside of set qc ranges.

Interesting that all failures I have seen have the given time as 0.5.  In each case, the time range is a different pair of values.

The list of 19:
Calb2-IRES-Cre;Sst-IRES-FlpO;Ai65-460695.04.02.01
Cck-IRES-Cre;Vip-IRES-FlpO;Ai65-417711.02.01.01
Crh-IRES-Cre_ZJH;Sst-IRES-FlpO;Ai65-454284.03.02.01
Gad2-IRES-Cre;Ai14-393692.04.02.01
H18.03.317.12.16.01.07
H18.03.322.11.11.01.02
H18.06.363.11.06.02.01
H18.06.367.11.13.01.10
Pvalb-IRES-Cre;Ai14-400569.04.02.02
Pvalb-IRES-Cre;Ai14-400570.04.02.01
Pvalb-IRES-Cre;Ai14-440449.04.02.03
Pvalb-T2A-CreERT2;Ai14-396262.04.02.02
Slc17a6-IRES-Cre;Ai14-490922.04.02.01
Sncg-IRES2-FlpO-neo;Ai65F-479903.05.01.02
Sst-IRES-Cre;Ai14-394301.05.01.02
Sst-IRES-Cre;Ai14-402666.04.02.01
Sst-IRES-Cre;Ai14-408419.03.01.02
Th-P2A-FlpO;Ai65F-391834.04.01.01
Vipr2-IRES2-Cre;Pvalb-T2A-FlpO;Ai65-462281.02.01.01

```
Traceback (most recent call last):
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/bin/run_pipeline_from_nwb_file.py", line 59, in <module>
    main()
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/bin/run_pipeline_from_nwb_file.py", line 53, in main
    pipeline_input["manual_sweep_states"])
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/bin/run_pipeline.py", line 59, in run_pipeline
    se_output['cell_features'],
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/bin/run_feature_extraction.py", line 52, in run_feature_extraction
    cell_features, sweep_features, cell_record, sweep_records = dsft.extract_data_set_features(data_set)
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/data_set_features.py", line 314, in extract_data_set_features
    subthresh_min_amp)
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/data_set_features.py", line 220, in extract_cell_features
    ramp_features = ramp_an.analyze(ramp_sweeps)
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/stimulus_protocol_analysis.py", line 93, in analyze
    features = super(RampAnalysis, self).analyze(sweep_set)
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/stimulus_protocol_analysis.py", line 84, in analyze
    self.analyze_basic_features(sweep_set, extra_sweep_features=extra_sweep_features, exclude_clipped=exclude_clipped)
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/stimulus_protocol_analysis.py", line 74, in analyze_basic_features
    self._spikes_set.append(self.spx.process(sweep.t, sweep.v, sweep.i))
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/feature_extractor.py", line 93, in process
    dvdt=dvdt)
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/spike_detector.py", line 39, in detect_putative_spikes
    start_index = tsu.find_time_index(t, start)
  File "/allen/aibs/technology/waynew/conda/ipfx37_dev/lib/python3.7/site-packages/ipfx/time_series_utils.py", line 20, in find_time_index
    assert t[0] <= t_0 <= t[-1], "Given time ({:f}) is outside of time range ({:f}, {:f})".format(t_0, t[0], t[-1])
AssertionError: Given time (0.500000) is outside of time range (-0.522500, 0.436200)
```

- [x] Identify the cause of the issue
- [x] Make code changes
- [x] Update the integration tests that were changed as a result of change in of the start time
- [x] Move integration tests to git-lfs
- [ ] Enable git-lfs on Bamboo build plan

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -54,7 +54,6 @@ jobs:
       - run:
           name: run tests
           command: |
-            sudo apt-get -y install hdf5-tools
             . venv/bin/activate
             mkdir test-results
             curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash

--- a/.gitattributes
+++ b/.gitattributes
@@ -1,3 +1,3 @@
 docs/gallery/data/* filter=lfs diff=lfs merge=lfs -text
-data/** filter=lfs diff=lfs merge=lfs -text
+tests/data/** filter=lfs diff=lfs merge=lfs -text
 

--- a/ipfx/bin/plot_ephys_nwb.py
+++ b/ipfx/bin/plot_ephys_nwb.py
@@ -1,7 +1,7 @@
 import sys
 import matplotlib.pyplot as plt
 import numpy as np
-from ipfx.data_set_utils import create_data_set
+from ipfx.dataset.create import create_ephys_data_set
 from ipfx.stimulus import StimulusOntology
 import allensdk.core.json_utilities as ju
 
@@ -89,6 +89,7 @@ def get_vertical_offset(data):
 def main():
 
     """
+    Plot sweeps of a given ephys nwb file
     # Usage:
     $ python plot_ephys_nwb_file.py NWB_FILE_NAME
 
@@ -100,12 +101,12 @@ def main():
     stimulus_ontology_file = StimulusOntology.DEFAULT_STIMULUS_ONTOLOGY_FILE
     ont = StimulusOntology(ju.read(stimulus_ontology_file))
 
-    data_set = create_data_set(nwb_file=nwb_file,validate_stim=False,ontology=ont)
-
-    vclamp_sweep_table = data_set.sweep_table[data_set.sweep_table["clamp_mode"] == "VoltageClamp"]
+    data_set = create_ephys_data_set(nwb_file=nwb_file)
+    vclamp_sweep_table = data_set.filtered_sweep_table(clamp_mode=data_set.VOLTAGE_CLAMP)
     plot_data_set(data_set, vclamp_sweep_table, nwb_file)
 
-    iclamp_sweep_table = data_set.sweep_table[data_set.sweep_table["clamp_mode"] == "CurrentClamp"]
+    data_set = create_ephys_data_set(nwb_file=nwb_file)
+    iclamp_sweep_table = data_set.filtered_sweep_table(clamp_mode=data_set.CURRENT_CLAMP)
     plot_data_set(data_set, iclamp_sweep_table, nwb_file)
 
     plt.show()

--- a/ipfx/feature_extractor.py
+++ b/ipfx/feature_extractor.py
@@ -265,7 +265,7 @@ class SpikeTrainFeatureExtractor(object):
         features = strf.basic_spike_train_features(t, spikes_df, self.start, self.end, exclude_clipped=exclude_clipped)
 
         if self.start is None:
-            self.start = t[0]
+            self.start = 0.0
 
         if extra_features is None:
             extra_features = []

--- a/tests/.gitignore
+++ b/tests/.gitignore
@@ -1,4 +1,3 @@
 reference_*
-*.nwb
 *.hdf5
 test-reports

--- a/tests/data/.gitattributes
+++ b/tests/data/.gitattributes
@@ -1,3 +1,0 @@
-version https://git-lfs.github.com/spec/v1
-oid sha256:51dfc6b23e1bbcc26d7b72fc8a3a44796e9e2723128d376758d7547538df09c9
-size 156

--- a/tests/data/nwb/Ctgf-T2A-dgCre;Ai14-495723.05.02.01-compressed-V2.nwb
+++ b/tests/data/nwb/Ctgf-T2A-dgCre;Ai14-495723.05.02.01-compressed-V2.nwb
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:ed8c8ed0af8cdd140efb142ff6f32f11665b40fa1ec3dd652c2cd6740b82289a
+size 21877490

--- a/tests/data/nwb/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02-compressed-V2.nwb
+++ b/tests/data/nwb/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02-compressed-V2.nwb
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:69cd2ed3611998705fe819b97ff6adb03f7c12d0e6a8face149f13ba6d7e4387
+size 170150361

--- a/tests/data/nwb/Vip-IRES-Cre;Ai14-331294.04.01.01-compressed-V2.nwb
+++ b/tests/data/nwb/Vip-IRES-Cre;Ai14-331294.04.01.01-compressed-V2.nwb
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:5d9dcb918bdfb60a79062890a3b77a58b26b83ca27aef473747c4daf377825d6
+size 17332553

--- a/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_input.json
+++ b/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_input.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:da35cc60a287edbc0c251fa372353a8f818910ff80ebe95de57496465f5d5c7b
+size 966

--- a/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_output.json
+++ b/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_output.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:bbd9900ecaf28e2f90d402b00ac562c97aae650fd8286431d4e016540597794f
+size 379115

--- a/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_output_nwb1.json
+++ b/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_output_nwb1.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:31836876e7d34299ad26bab5e7f675bf2dbda486c07c27fa5787040032275038
+size 383066

--- a/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/stimulus_ontology.json
+++ b/tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/stimulus_ontology.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:62ce5ed148c99b343e672d06364c433ec29046c58ea5602cbac5679650bb3af9
+size 24526

--- a/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_input.json
+++ b/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_input.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:44388f0c2ceb451bf35a5dd4926216c2d80c07fa851934d106321d38e7687ee7
+size 974

--- a/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_output.json
+++ b/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_output.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:9d576bc187f9158caf3129da8f36f7e5c7e1470f72f7065984271a606c67dc0a
+size 1652116

--- a/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_output_nwb1.json
+++ b/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_output_nwb1.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:f41c0083e62f3f1ba0c493ad13117061030f08c5522705be65def0b8b5729b77
+size 1667810

--- a/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/stimulus_ontology.json
+++ b/tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/stimulus_ontology.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:62ce5ed148c99b343e672d06364c433ec29046c58ea5602cbac5679650bb3af9
+size 24526

--- a/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_input.json
+++ b/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_input.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:ffe25ad97e7764cccea1219a27b11cf926d21b92f2d1cccb26411b9d3ae3dfca
+size 962

--- a/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output.json
+++ b/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:c17159af752cb886812faf28cbf41d83fef274ed108fdde45403c00b32b6aa7c
+size 439843

--- a/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output_nwb1.json
+++ b/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output_nwb1.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:90ad6aaf33b89bdbbb464f7fa2311279268c5460bd32ed4395a1eed1e40ac602
+size 441867

--- a/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output_nwb2.json
+++ b/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output_nwb2.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:c40ce5ee6889fd27248957125c810ed0944704a827ec3f2ff16325e6943f062f
+size 446639

--- a/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/stimulus_ontology.json
+++ b/tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/stimulus_ontology.json
@@ -1,0 +1,3 @@
+version https://git-lfs.github.com/spec/v1
+oid sha256:62ce5ed148c99b343e672d06364c433ec29046c58ea5602cbac5679650bb3af9
+size 24526

--- a/tests/test_mies_nwb2_specimens.csv
+++ b/tests/test_mies_nwb2_specimens.csv
@@ -1,4 +1,4 @@
 input_json output_json
-/allen/aibs/informatics/module_test_data/ipfx/test_specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_input_nwb2.json /allen/aibs/informatics/module_test_data/ipfx/test_specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_output_nwb1.json
-/allen/aibs/informatics/module_test_data/ipfx/test_specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_input_nwb2.json /allen/aibs/informatics/module_test_data/ipfx/test_specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output_nwb1.json
-/allen/aibs/informatics/module_test_data/ipfx/test_specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_input_nwb2.json /allen/aibs/informatics/module_test_data/ipfx/test_specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_output_nwb1.json
+./tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_input.json ./tests/data/specimens/Pvalb-IRES-Cre;Ai14(IVSCC)-165172.05.02/pipeline_output.json
+./tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_input.json ./tests/data/specimens/Vip-IRES-Cre;Ai14-331294.04.01.01/pipeline_output.json
+./tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_input.json ./tests/data/specimens/Ctgf-T2A-dgCre;Ai14-495723.05.02.01/pipeline_output.json

--- a/tests/test_mies_nwb_pipeline_output.py
+++ b/tests/test_mies_nwb_pipeline_output.py
@@ -41,7 +41,6 @@ def rebase(new_base: str, path: str, mkdir: bool = False) ->  str:
     return str(new_path)
 
 
-@pytest.mark.requires_inhouse_data
 @pytest.mark.slow
 @pytest.mark.parametrize('input_json,output_json', test_specimens_params)
 def test_mies_nwb_pipeline_output(input_json, output_json, tmpdir_factory):
@@ -66,8 +65,6 @@ def test_mies_nwb_pipeline_output(input_json, output_json, tmpdir_factory):
     pipeline_input["qc_figs_dir"] = None
 
     stimulus_ontology_file = pipeline_input.get("stimulus_ontology_file", None)
-    if stimulus_ontology_file is not None:
-        stimulus_ontology_file = rebase(test_dir, stimulus_ontology_file, True)
 
     obtained = run_pipeline(
         pipeline_input["input_nwb_file"],

